yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+
+    }, {
+        'url': 'https://www.nytimes.com/2023/12/02/business/media/netflix-squid-game-challenge.html',
         'only_matching': True,
     }]
 
-    def _extract_podcast_from_json(self, json, page_id, webpage):
-        podcast_audio = self._parse_json(
-            json, page_id, transform_source=js_to_json)
+    def _extract_content_from_block(self, block):
+        details = traverse_obj(block, {
+            'id': ('sourceId', {str}),
+            'uploader': ('bylines', ..., 'renderedRepresentation', {str}),
+            'duration': (('duration', 'length', ...), {int_or_none}),
+            'upload_date': ('firstPublished', {unified_strdate}),
+            'timestamp': ('firstPublished', {parse_iso8601}),
+            'series': ('podcastSeries', {str})}, get_all=False)
+
+        media = traverse_obj(block, ('renditions', ...))
+        formats, subtitles = self._extract_media_from_json(details.get('id'), media)
+
+        # audio articles will have an url and no formats
+        url = traverse_obj(block, ('fileUrl'), get_all=False)
+        if not formats and url:
+            formats.append({'url': url, 'ext': determine_ext(url)})
+
+        return merge_dicts(
+            details,
+            {
+                'formats': formats,
+                'subtitles': subtitles
+            })
 
-        audio_data = podcast_audio['data']
-        track = audio_data['track']
+    def _extract_thumbnails(self, thumbs):
+        thumbnails = []
+        for image in thumbs:
+            image_url = image.get('url')
+            if not image_url:
+                continue
+            thumbnails.append({
+                'url': image_url,
+                'width': int_or_none(image.get('width')),
+                'height': int_or_none(image.get('height')),
+            })
+        return thumbnails
 
-        episode_title = track['title']
-        video_url = track['source']
+    def _real_extract(self, url):
+        page_id = self._match_id(url)
+        webpage = self._download_webpage(url, page_id)
+        article_json = self._search_json(
+            r'window\.__preloadedData\s*=', webpage, 'media details', page_id,
+            transform_source=lambda x: x.replace('undefined', 'null'))['initialData']['data']['article']
+        blocks_json = traverse_obj(
+            article_json, ('sprinkledBody', 'content', ..., ('ledeMedia', None), lambda _, v: v['__typename'] in ('Video', 'Audio')))
+
+        art_title = remove_end(self._html_extract_title(webpage), ' - The New York Times')
+        art_description = traverse_obj(
+            article_json, ('sprinkledBody', 'content', ..., 'summary', 'content', ..., 'text'),
+            get_all=False) or self._html_search_meta(['og:description', 'twitter:description'], webpage)
+        art_upload_date = traverse_obj(article_json, ('firstPublished'))
+        creator = ', '.join(
+            traverse_obj(article_json, ('bylines', ..., 'creators', ..., 'displayName'), get_all=True))
```

Comment:
`get_all=True` is the default
```suggestion
            traverse_obj(article_json, ('bylines', ..., 'creators', ..., 'displayName')))
```